File: examples/js/basic.js
/**
 * CommonJS style module (must be .js)
 */

const RES_FILTER = "(and true)"
const REQ_FILTER = "(and true)"

const HttpParameter = Java.type("burp.api.montoya.http.message.params.HttpParameter")
const HttpParameterType = Java.type("burp.api.montoya.http.message.params.HttpParameterType")

function initialize() {
    log.info("Initialized the JavaScript module");
}

function cleanup() {
    log.info("Cleaning up JavaScript");
}

function onRequest(req) {
    log.info(`${req.method()} - ${req.url()}`)
    return req.withParameter(
        HttpParameter.parameter("__ivision", "injected", HttpParameterType.URL)
    )
}

function onResponse(res) {
    log.info(`${res.statusCode()} - ${res.reasonPhrase()}`);
    return res;
}

module.exports = {
    RES_FILTER,
    REQ_FILTER,
    initialize,
    cleanup,
    onRequest,
    onResponse
}

File: examples/python/basic.py
import java

HttpParameter = java.type("burp.api.montoya.http.message.params.HttpParameter")
HttpParameterType = java.type("burp.api.montoya.http.message.params.HttpParameterType")

REQ_FILTER = """
(and
    (in-scope)
    (path-contains "api")
)
"""

RES_FILTER = """
(header-matches "Content-Type" r"application/json")
"""

def initialize():
    log.info("Initialized Python script")


def cleanup():
    log.info("Cleaning up Python script")


def on_request(req):
    log.info(f"{req.method()} - {req.url()}")
    return req.withParameter(
        HttpParameter.parameter("__ivision", "injected", HttpParameterType.URL)
    )


def on_response(res):
    log.info(f"{res.statusCode()} - {res.reasonPhrase()}")

